clang/test/SemaCXX/coroutine-unreachable-warning.cpp

```diff
@@ -12,6 +12,13 @@ struct task {
     std::suspend_always yield_value(int) { return {}; }
     task get_return_object();
     void unhandled_exception();
+
+    struct Awaiter {
+      bool await_ready();
+      void await_suspend(auto);
+      int await_resume();
+    };
+    auto await_transform(const int& x) { return Awaiter{}; }
```

Comment:
nit: (Just to reduce the boilerplate not related to the test itself) we do not need to define a new awaitable here. Maybe just:
`std::suspend_always await_transform(int) { return {}; }`

Reply:
We have a testcase `int a = co_await 1;` where the return type of `await_resume` needs to be `intt`, `std::suspend_always` is not sufficient enough for it.

Reply:
Oh my bad. I thought this was targeting `co_await some_integer;`. Makes sense then.